internal/job/testutil/testutil.go

```diff
@@ -92,17 +98,38 @@ func CreateNLVAndFinRepoForTest(t *testing.T) (*nlv.NodeLocalVolumeRepository, m
 	return nlvRepo, repo, tempDir
 }
 
-func CreateRestoreFileForTest(t *testing.T, size uint64) string {
+func CreateLoopDevice(t *testing.T, size uint64) string {
 	t.Helper()
 
-	restoreFile, err := os.CreateTemp("", "fake-restore-*.img")
+	loopbackFile, err := os.CreateTemp("", "fake-restore-*.img")
 	require.NoError(t, err)
-	defer func() { _ = restoreFile.Close() }()
-	restorePath := restoreFile.Name()
-	t.Cleanup(func() { _ = os.Remove(restorePath) })
-	err = restoreFile.Truncate(int64(size))
+	defer func() { _ = loopbackFile.Close() }()
+
+	err = loopbackFile.Truncate(int64(size))
 	require.NoError(t, err)
-	return restorePath
+
+	loopbackFilePath := loopbackFile.Name()
+	_, err = executeSudo("losetup", "-f", loopbackFilePath)
+	require.NoError(t, err, "failed to create loop device")
+	// get loop device path
+	stdout, err := executeSudo("losetup", "-j", loopbackFilePath)
+	require.NoError(t, err, "failed to get loop device path")
+	devicePath := strings.Split(string(stdout), ":")[0]
+	require.NotEmpty(t, devicePath, "loop device path should not be empty")
+	_, err = executeSudo("chmod", "666", devicePath)
+	require.NoError(t, err, "failed to change permissions of loop device")
+
+	t.Cleanup(func() {
```

Comment:
I think `t.Cleanup` should be called immediately after its corresponding process, that is,

```go
	loopbackFile, err := os.CreateTemp("", "fake-restore-*.img")
	require.NoError(t, err)
	t.Cleanup(func() { _ = os.Remove(loopbackFile.Name()) }

	/* ... */

	_, err = executeSudo("losetup", "-f", loopbackFilePath)
	require.NoError(t, err, "failed to create loop device")
	t.Cleanup(func() { /* ... */ })

	/* ... */

	_, err = executeSudo("chmod", "666", devicePath)
	require.NoError(t, err, "failed to change permissions of loop device")
	t.Cleanup(func() { /* ... */ })
```

because cleaning up process should be run even when errors occur in between.